package.json

```diff
@@ -27,32 +27,56 @@
     "package.json",
     "prettier"
   ],
-  "bin": "lib/cli/index.js",
-  "main": "lib/index.js",
+  "bin": "build/cli/index.js",
+  "main": "build/lib/index.js",
+  "files": [
+    "build",
+    "yarn.lock",
+    "LICENSE",
+    "README.md",
+    "CHANGELOG.md"
+  ],
   "scripts": {
-    "docs": "scripts/markdown.js",
-    "format": "npm-run-all format:*",
-    "format:docs": "prettier --write '**/*.md'",
-    "format:package": "./lib/cli/index.js -w package.json",
-    "format:source": "prettier --write '**/*.{js,json}'",
-    "gamut": "npm-run-all reset docs format lint test",
-    "lint": "eslint \"**/*.js\"",
+    "prebuild": "yarn run clean-build",
+    "build": "tsc",
+    "postbuild": "chmod +x ./build/cli/index.js",
+    "clean": "npm-run-all clean-build clean-packages",
+    "clean-build": "rimraf ./build",
+    "clean-packages": "rimraf ./node_modules",
+    "dev": "ts-node-dev --respawn --transpileOnly -- ./src/cli/index.ts",
+    "docs": "ts-node ./scripts/markdown.ts",
+    "format": "npm-run-all format-package format-source",
+    "format-docs": "prettier --write '**/*.md'",
+    "format-package": "ts-node ./src/cli/index.ts -w package.json",
+    "format-source": "prettier \"**/**{js,json,ts}\" --write",
+    "gamut": "npm-run-all reset format lint type-check test build start",
+    "lint": "eslint . --ext .ts,.js",
     "prepublishOnly": "yarn gamut",
-    "reset": "rimraf node_modules && yarn install",
-    "test": "jest"
+    "reset": "yarn run clean && yarn install",
+    "start": "node ./build/cli/index.js",
+    "test": "jest",
+    "type-check": "tsc --noEmit",
+    "watch": "ts-node-dev --respawn --transpileOnly"
+  },
+  "husky": {
+    "hooks": {
+      "post-merge": "install-deps-postmerge",
+      "pre-commit": "lint-staged",
+      "pre-push": "yarn test --coverage"
+    }
   },
   "lint-staged": {
     "**/package.json": [
-      "yarn format:package",
+      "yarn format-package",
       "git add"
     ],
-    "*.js": [
+    "*.json": [
       "prettier --write",
-      "eslint",
       "git add"
     ],
-    "*.json": [
+    "*.{js,ts}": [
       "prettier --write",
+      "eslint",
```

Comment:
Should change this to `yarn lint` for `--ext`.

Reply:
When feeding it the list of files directly, `ext` does not seem needed - e.g.

```sh
yarn eslint src/lib/index.ts
```